src/freedv_api.c

```diff
@@ -855,7 +855,7 @@ int freedv_bits_to_speech(struct freedv *f, short speech_out[], short demod_in[]
            }
            else {
                /* sync is solid - decode even through fades as there is still some speech info there */
-               if (f->stats.snr_est > f->snr_squelch_thresh)
+               if (f->snr_est > f->snr_squelch_thresh)
```

Comment:
Thanks @tmiw :+1: 

This same function is used by all voice modes, so we'll need to make sure other modes are not affected. So far we haven't had any reports of similar issues on 1600, 700C/D/E, or 2020. 

OK so it looks like the other modes set `f->snr` like [this example for 700C](https://github.com/drowe67/codec2/blob/master/src/freedv_700.c#L291)

I think `f->snr` is set [here for 800XA](https://github.com/drowe67/codec2/blob/master/src/freedv_fsk.c#L575).
 
BTW the code above and the FSK modem were refactored mid-2020, which might explain why the bug just appeared.  Suspect it's an Eb/No to SNR scaling thing.  I'll look into it some more.

Reply:
@drowe67, I did some brief testing with 700D and squelch still seemed to work, FWIW. However, perhaps we can see where SNR should have gotten copied to f->stats for 800XA and make sure that happens instead?

Reply:
@tmiw IIIRC my general design idea was for all modes to set  `f->snr` during the freedv_rx() call, so the logic in `freedv_bits_to_speech()` can then handle squelch for all modes. The way the SNR is determined could vary across modems (e.g. some modems might not fill the stats structure). `f->stats` isn't actually filled in `freedv_comprx_fsk()` although I think it gets filled later when we grab extended info for the eye diagram - which might be why the initial fix happens to work.

I'll look into the SNR calculation for 800XA.

Reply:
@tmiw Ahh Ok I misread the initial patch above - yes I agree with your fix.  As per the comment above it's what I had intended in the design. Good work :100:

Reply:
@tmiw I'm running a few tests on squelch using the sample 700D file and the built-in Tools-Options noise generator.  However I'm getting some odd results, e.g. when I enable the noise, and press Resync : 700D won't resync properly and the audio stops.  This is odd, as I haven't had any reports of this behavior from off air signals.  Could be a configuration issue on my end.

I've run out of time to work on this today, will try again later this week.

Reply:
@drowe67, FWIW, I see the same issue using full duplex too, not via playing back the sample file (700D and 700E, haven't tried other modes). However, it doesn't seem related to this fix as I still see it even after backing out the change. Not sure what it could be offhand.

Reply:
@drowe67, I noticed a comment in freedv_set_sync() indicating that it needed to be called from the same thread that calls freedv_rx(). I made an update so that pushing the button sets a global variable that's then checked in txRxProcessing() before calling freedv_set_sync() and that seems to have fixed the issue for me. Should I check that into the same bugfixes branch as the scatter plot changes, or is a different branch/PR preferred?

Reply:
@tmiw thanks for looking into that.  Lol looks like I should read my own documentation!  Yes I agree it's unrelated to the changes in this PR.  Adding your change to the freedv bug fix PR would be great, thanks :+1:

Reply:
@drowe67, done. See drowe67/freedv-gui#85.